src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerDataSource.java

```diff
@@ -424,4 +497,297 @@
+
+    /**
+     * Retrieves the Always Encrypted functionality setting for the data source object.
+     * 
+     * @return the Always Encrypted functionality setting for the data source object.
+     */
+    public String getColumnEncryptionSetting();
+
+    /**
+     * Sets the name that identifies a key store. Only value supported is the "JavaKeyStorePassword" for identifying the Java Key Store. The default
+     * is null.
+     * 
+     * @param keyStoreAuthentication
+     *            the name that identifies a key store.
+     */
+    public void setKeyStoreAuthentication(String keyStoreAuthentication);
+
+    /**
+     * Gets the value of the keyStoreAuthentication setting for the data source object.
+     * 
+     * @return the value of the keyStoreAuthentication setting for the data source object.
+     */
+    public String getKeyStoreAuthentication();
+
+    /**
+     * Sets the password for the Java keystore. Note that, for Java Key Store provider the password for the keystore and the key must be the same.
+     * Note that, keyStoreAuthentication must be set with "JavaKeyStorePassword".
+     * 
+     * @param keyStoreSecret
+     *            the password to use for the keystore as well as for the key
+     */
+    public void setKeyStoreSecret(String keyStoreSecret);
+
+    /**
+     * Sets the location including the file name for the Java keystore. Note that, keyStoreAuthentication must be set with "JavaKeyStorePassword".
+     * 
+     * @param keyStoreLocation
+     *            the location including the file name for the Java keystore.
+     */
+    public void setKeyStoreLocation(String keyStoreLocation);
+
+    /**
+     * Retrieves the keyStoreLocation for the Java Key Store.
+     * 
+     * @return the keyStoreLocation for the Java Key Store.
+     */
+    public String getKeyStoreLocation();
+
+    /**
+     * Setting the query timeout
+     * 
+     * @param queryTimeout
+     *            The number of seconds to wait before a timeout has occurred on a query. The default value is 0, which means infinite timeout.
+     */
+    public void setQueryTimeout(int queryTimeout);
+
+    /**
+     * Getting the query timeout
+     * 
+     * @return The number of seconds to wait before a timeout has occurred on a query.
+     */
+    public int getQueryTimeout();
+
+    /**
+     * Setting the cancel timeout
+     * 
+     * @param cancelQueryTimeout
+     *            The number of seconds to wait before we wait for the query timeout to happen.
+     */
+    public void setCancelQueryTimeout(int cancelQueryTimeout);
+
+    /**
+     * Getting the cancel timeout
+     * 
+     * @return the number of seconds to wait before we wait for the query timeout to happen.
+     */
+    public int getCancelQueryTimeout();
+
+    /**
+     * If this configuration is false the first execution of a prepared statement will call sp_executesql and not prepare a statement, once the second
```

Comment:
start with description as in other methods?
ie This setting controls blah blah...